src/main/java/org/nguh/nguhcraft/data/NguhcraftRecipeGenerator.kt

```diff
@@ -226,6 +226,12 @@ class NguhcraftRecipeGenerator(
 
         offerShapelessRecipe(NguhBlocks.CINNABAR, 2, Items.NETHERRACK to 1, Items.COBBLESTONE to 1)
 
+        offerShaped(Items.TUFF, 4) {
+            pattern("BQ")
+            pattern("QB")
+            cinput('B', Items.BLACKSTONE)
+            cinput('Q', Items.QUARTZ)
+        }
```

Comment:
```suggestion

        offerShaped(Items.TUFF, 4) {
            pattern("BQ")
            pattern("QB")
            cinput('B', Items.BLACKSTONE)
            cinput('Q', Items.QUARTZ)
        }
        
        offerShapelessRecipe(NguhBlocks.CINNABAR, 2, Items.NETHERRACK to 1, Items.COBBLESTONE to 1)
        
```

Reply:
Put shaped recipes before shapless recipes and leave an empty line between sections